### PR TITLE
Empty arrays are meaningful

### PR DESCRIPTION
**Pull request by thisthat**
## Changes

Stress that empty arrays must be forwarded to span processors/exporters.

### CODE REVIEW

**Comment by Oberon00** [comment]
I think this should also apply to resource attributes and thus be moved to the common attribute spec.

**Comment by carlosalberto** [comment]
This makes sense to me. Let's wait for other's feedback ;)

**Comment by carlosalberto** [comment]
Btw, since `Attributes` are used in other places, maybe it makes sense to review and maybe updates other parts of the Specification?

**Comment by jkwatson** [comment]
> I think this should also apply to resource attributes and thus be moved to the common attribute spec.

Let's not muddy this PR with a discussion of Resource attributes. I recommend creating an issue or a PR to deal with that, independently.

**Comment by carlosalberto** [comment]
> Let's not muddy this PR with a discussion of Resource attributes

Fair enough, will create a follow up issue.

File: specification/trace/api.md [hunk 1]
```diff
@@ -406,7 +406,7 @@ Attributes SHOULD preserve the order in which they're set. Setting an attribute
 with the same key as an existing attribute SHOULD overwrite the existing
 attribute's value.
 
-Attribute values expressing a numerical value of zero or an empty string are
+Attribute values expressing a numerical value of zero, an empty string, or an empty array are
```
Comment (bogdandrutu) [inline specification/trace/api.md:409]: Should this apply to all attributes? If yes probably we should update https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/common/common.md and remove this.

**Pull request merged**